commons/all/src/main/java/org/infinispan/commons/configuration/io/xml/MXParser.java

```diff
@@ -6,78 +6,86 @@
 import java.io.InputStreamReader;
 import java.io.Reader;
 import java.io.UnsupportedEncodingException;
-import java.nio.charset.StandardCharsets;
+import java.util.Objects;
 
 /**
  * Absolutely minimal implementation of XMLPULL V1 API
  *
  * @author <a href="http://www.extreme.indiana.edu/~aslom/">Aleksander Slominski</a>
  */
 
-public class MXParser implements XmlPullParser {
-   protected final static String XML_URI = "http://www.w3.org/XML/1998/namespace";
-   protected final static String XMLNS_URI = "http://www.w3.org/2000/xmlns/";
-   protected final static String FEATURE_XML_ROUNDTRIP =
+public class MXParser
+      implements XmlPullParser {
+   //NOTE: no interning of those strings --> by Java lang spec they MUST be already interned
+   private final static String XML_URI = "http://www.w3.org/XML/1998/namespace";
+   private final static String XMLNS_URI = "http://www.w3.org/2000/xmlns/";
+   private final static String FEATURE_XML_ROUNDTRIP =
+         //"http://xmlpull.org/v1/doc/features.html#xml-roundtrip";
          "http://xmlpull.org/v1/doc/features.html#xml-roundtrip";
-   protected final static String FEATURE_NAMES_INTERNED =
+   private final static String FEATURE_NAMES_INTERNED =
          "http://xmlpull.org/v1/doc/features.html#names-interned";
-   protected final static String PROPERTY_XMLDECL_VERSION =
+   private final static String PROPERTY_XMLDECL_VERSION =
          "http://xmlpull.org/v1/doc/properties.html#xmldecl-version";
-   protected final static String PROPERTY_XMLDECL_STANDALONE =
+   private final static String PROPERTY_XMLDECL_STANDALONE =
          "http://xmlpull.org/v1/doc/properties.html#xmldecl-standalone";
-   protected final static String PROPERTY_XMLDECL_CONTENT =
+   private final static String PROPERTY_XMLDECL_CONTENT =
          "http://xmlpull.org/v1/doc/properties.html#xmldecl-content";
-   protected final static String PROPERTY_LOCATION =
+   private final static String PROPERTY_LOCATION =
          "http://xmlpull.org/v1/doc/properties.html#location";
 
    /**
-    * Implementation notice: the is instance variable that controls if newString() is interning.
+    * Implementation notice:
+    * the is instance variable that controls if newString() is interning.
     * <p><b>NOTE:</b> newStringIntern <b>always</b> returns interned strings
     * and newString MAY return interned String depending on this variable.
     * <p><b>NOTE:</b> by default in this minimal implementation it is false!
     */
-   protected boolean allStringsInterned;
+   private final boolean allStringsInterned = false;
 
-   protected String newString(char[] cbuf, int off, int len) {
+   private void resetStringCache() {
+      //System.out.println("resetStringCache() minimum called");
```

Comment:
Empty method and we have commented code all over the place. Do we need to keep those?

Reply:
I didn't want to deviate too much from the original